src/proof_system/proof.rs

```diff
@@ -491,11 +493,10 @@ fn compute_barycentric_eval<F: PrimeField>(
         .collect();
 
     // Only compute the denominators with non-zero evaluations
-    let range = (0..non_zero_evaluations.len()).into_iter();
+    let range = 0..non_zero_evaluations.len();
 
-    let group_gen_inv = util::get_domain_attrs(&domain, "group_gen")
-        .inverse()
-        .unwrap();
+    // FIXME: Should we just call `domain.group_gen_inv()` here?
```

Comment:
Yes, that's better

Reply:
I don't agree. Until the PR we made to arkworks does not land, I'd not mix the two ways of getting data from the struct TBH. I know the `util::` is ugly, but if we have the two ways at the same time makes it really confusing.

Reply:
In any case, we should use `group_gen_inv` and remove the `inverse()` call